src/replication.c

```diff
@@ -373,14 +460,18 @@ void feedReplicationBuffer(char *s, size_t len) {
             server.primary_repl_offset += copy;
             server.repl_backlog->histlen += copy;
         }
+        if (empty_backlog && raxSize(server.replicas_waiting_psync) > 0) {
+            /* Increase refcount for pending replicas. */
+            addReplicaToPsyncWaitingRaxRetrospect();
+        }
```

Comment:
Can this code be hit? I am not sure when we could miss registering a replica for psync-wait?

Reply:
This issue can occur and has been reproduced in the test case `Test rdb-connection connection peering - start with empty backlog (retrospect)`. 

The scenario unfolds as follows:
1. Initially, the master has an empty backlog when the replica begins synchronization using an RDB connection. This empty state occurs because the master clears its entire backlog after a period of inactivity.
2. The replica then starts receiving the RDB data.
3. Before the replica can initiate a PSYNC operation, other clients execute write commands on the master.
4. These write operations cause the master's backlog to increase, creating the first block of the the replication buffer.

Reply:
Got it. This is the case where we start with 0 data in the shard.